openmc/deplete/nuclide.py

```diff
@@ -211,14 +213,190 @@ def to_xml_element(self):
             fpy_elem = ET.SubElement(elem, 'neutron_fission_yields')
             energy_elem = ET.SubElement(fpy_elem, 'energies')
             energy_elem.text = ' '.join(str(E) for E in self.yield_energies)
+            self.yield_data.to_xml_element(fpy_elem)
 
-            for E in self.yield_energies:
-                yields_elem = ET.SubElement(fpy_elem, 'fission_yields')
-                yields_elem.set('energy', str(E))
+        return elem
 
-                products_elem = ET.SubElement(yields_elem, 'products')
-                products_elem.text = ' '.join(x[0] for x in self.yield_data[E])
-                data_elem = ET.SubElement(yields_elem, 'data')
-                data_elem.text = ' '.join(str(x[1]) for x in self.yield_data[E])
 
-        return elem
+class FissionYieldDistribution(Mapping):
+    """Class for storing energy-dependent fission yields for a single nuclide
+
+    Can be used as a dictionary mapping energies and products to fission
+    yields::
+
+        >>> fydist = FissionYieldDistribution.from_dict({
+        ...     {0.0253: {"Xe135": 0.021}})
+        >>> fydist[0.0253]["Xe135"]
+        0.021
+
+    Parameters
+    ----------
+    ordered_energies : iterable of float
+        Energies for which fission yield data exist
+    orderded_products : iterable of str
+        Fission products produced by this parent at all energies
+    group_fission_yields : numpy.ndarray or iterable of iterable of float
```

Comment:
Suggest renaming as `yield_matrix` since it is effectively the same as the attribute